File: src/main/java/org/khun/leetcode/TwoPointers/IsSubsequence.java
package org.khun.leetcode.TwoPointers;

public class IsSubsequence {

    public static boolean isSubsequence(String s, String t) {
        int pointerS = 0;
        int pointerT = 0;
        while (pointerS < s.length() && pointerT < t.length()){
            if (s.charAt(pointerS) == t.charAt(pointerT)){
                pointerS++;
            }
            pointerT++;
        }

        return pointerS == s.length();
    }

    public static void main(String[] args) {
        System.out.println(isSubsequence("abc", "ahbgdc"));
        System.out.println(isSubsequence("axc", "ahbgdc"));
        System.out.println(isSubsequence("bb", "ahbgdc"));
        System.out.println(isSubsequence("ab", "baab"));
    }
}

File: src/main/java/org/khun/leetcode/TwoPointers/IsSubsequence.kt
package org.khun.leetcode.TwoPointers

fun isSubsequence(s: String, t: String): Boolean {
    var pointerS = 0
    var pointerT = 0

    while (pointerS<s.length && pointerT<t.length){
        if (s[pointerS] == t[pointerT]) pointerS++
        pointerT++
    }
    return pointerS == s.length
}

fun main() {
    println(isSubsequence("abc", "ahbgdc"))
    println(isSubsequence("axc", "ahbgdc"))
    println(isSubsequence("bb", "ahbgdc"))
    println(isSubsequence("ab", "baab"))
}